content/blog/hackaton-2022-building-a-pong.md

```diff
@@ -0,0 +1,212 @@
+
+---
+Title: Hackapong fun at the shop!
+Date: 2022-11-04 13::55:00
+Author: Jean-Philippe Lamebrt - alex Hoguin
+Category: WDA
+Tags: hackathon, wda
+Slug: hackaton-2022-building-a-pong
+Status: published
+---
+## Here Comes the Hackapong 
+Our Yearly hackathon is a very important moment, first of all we use it to bond as a team and work as a team to use our tools and see what we could add in the future. We used this week to build on tools we already use or that are coming to WDA soon.
+
+Our team decided to put the WDA plugin functionnality to good use and we decided to build a pong game that would be playable between two callers using WDA. 
+
+To accomplish that feat we used existing tools from the platform, webrtc, asterisk, nginx, etc and some that are not yet available i.e. the plugin interface that is in the works WDA. 
+
+We started by finding a pong javascript [codebase](https://gist.github.com/straker/81b59eecf70da93af396f963596dfdc5) that was functional and had all the feature we needed. We used an open-source project and we built on that. 
+
+The issues we faced were mostly related to synchronisation and control of the paddles for the players. 
+
+## Intercommunication 
+
+Since we used DTMF's to control the paddles ( we wanted it to work with a plastic phone too ) we had to use the wazo-sdk javascript library. We linked the user control ( arrow up and arrow down ) to send DTMF's to the stack. 
+
+```javascript
+/**
+* When using the wazo-sdk we have directly acces to Wazo websocket events 
+* without too much work. 
+*/
+Wazo.Websocket.on('call_dtmf_created', ({ data }) => {
+
+	if (!data.user_uuid) {
+		return;
+	}
+
+	if (data.digit === '0' || data.digit === '1') {
+		const isPlayer1 = data.digit === '0';
+		return stopMovePlayer(isPlayer1);
+	}
+	/**
+	* Since we need to separate both player we sent different DTMF's
+	* Through the system.
+	*/
+	const isUp = data.digit === '8' || data.digit === '9';
+	const isPlayer1 = data.digit === '2' || data.digit === '8';
+	
+	movePlayer(isPlayer1, isUp);
+});
+
+```
+
+We also had to decide how to synchronise the ball between the two users, since we are not using a netcode library, we decided to make the caller the source of truth. One of the interesting part is that we are able to speak while playing the game. Oh and if you are very good at pong your calls wont be very long, we needed a way to make a winner so when you get to 20 points the game ends and the call hangs-up. 
```

Comment:
```suggestion
We also had to decide how to synchronise the ball between the two users. Since we are not using a netcode library, we decided to make the caller the source of truth. One of the interesting part is that we are able to speak while playing the game. Oh and if you are very good at pong your calls wont be very long, we needed a way to choose a winner so when you get to 20 points the game ends and the call hangs-up. 
```